application/client.plugins/projects/serial/src/lib/services/service.ts

```diff
@@ -204,8 +239,29 @@ export class Service extends Toolkit.APluginService {
         });
     }
 
-    public closePopup(popup: string) {
-        this.api.removePopup(popup);
+    public removePopup() {
+        this.api.removePopup(this._popupGuid);
+    }
+
+    public addPopup(popup: IPopup) {
+        this._popupGuid = this.api.addPopup(popup);
+    }
+
+    public notify(caption: string, message: string, type: ENotificationType) {
+        this.api.addNotification({
+            caption: caption,
+            message: message,
+            options: {
+                type: type
+            }
+        });
+        if (type === ENotificationType.error) {
+            this._logger.error(message);
+        } else if (type === ENotificationType.warning) {
+            this._logger.warn(message);
+        } else {
+            this._logger.info(message);
+        }
```

Comment:
You are using method `notify` outside (like `Service.notify`), but API could be available. So to prevent any errors (not available API), better add here checks for it, like `if (this.api === undefined) { ... }`